### PR TITLE
[sp] add right click context menu

### PR DESCRIPTION
**Pull request by shrey-mage**
# Summary
- build generic right click context menu for use throughout data prep tool
- add context menu to the file browser (currently, the only action supported is deleting block files)
- display error popup box if user tries to delete a block file with dependencies 

# Tests

cc: @johnson-mage

### CODE REVIEW

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:10]: You should probably change the name of the `context` prop to something else. Perhaps `location` or something else. Can be confused with React Context or a Nextjs context object.

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:32]: unnecessary parens inside the brackets

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:36]: unnecessary parens inside the brackets

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:40]: unnecessary parens inside the brackets

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:27]: Does simply using `props.left` instead of `typeof props.left !== 'undefined'` cause issues when `props.left` is 0 or some other issue?

File: mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx:94]: Did you make sure removal of `pauseEvent` doesn't reintroduce some other bug?

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (shrey-mage) [reply mage_ai/frontend/components/ContextMenu/index.tsx:28]: Yeah I think that's the reason why undefined is explicitly checked here. I actually copied this `ContainerStyle` from `PopupMenu`, but I assume that's why lol

File: mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx
Comment (shrey-mage) [reply mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx:94]: Will double check before merging

**Comment by johnson-mage** [comment]
The context menu with only a "Delete" menu seems a bit large. Can we reduce the width a bit?

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:16]: alphabetize

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:35]: alphabetize

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:91]: Could this use an empty dependency array for clarity?

File: mage_ai/frontend/components/FileBrowser/Folder.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/FileBrowser/Folder.tsx:18]: I'd probably include './index'

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:18]: Rename this to something else that doesn't start with `use`. Can get confused with a react hook.

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (johnson-mage) [inline mage_ai/frontend/components/ContextMenu/index.tsx:11]: This would presumably be optional if used in somewhere besides the file browser.

**Review by johnson-mage — APPROVED** [review]
approved after addressing comments

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (shrey-mage) [reply mage_ai/frontend/components/ContextMenu/index.tsx:91]: ESLint complains that it should include the two callback functions, which seems wrong, and suggests that I should remove the dep array if that isn't intended. Should I still add the empty array?

Comment (johnson-mage) [reply mage_ai/frontend/components/ContextMenu/index.tsx:91]: yeah, u shud still have a dep array

File: mage_ai/frontend/components/ContextMenu/index.tsx
Comment (shrey-mage) [reply mage_ai/frontend/components/ContextMenu/index.tsx:11]: oh good catch, thanks

File: mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx
Comment (shrey-mage) [reply mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx:94]: > The context menu with only a "Delete" menu seems a bit large. Can we reduce the width a bit?

@johnson-mage added a `compact` prop to `FlyoutMenu` that makes the min width `UNIT * 10`:

image

**Comment by shrey-mage** [comment]
> The context menu with only a "Delete" menu seems a bit large. Can we reduce the width a bit?

@johnson-mage added a `compact` prop to `FlyoutMenu` that makes the min width `UNIT * 20`:

image

File: mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx
Comment (shrey-mage) [reply mage_ai/frontend/components/PipelineDetail/KernelStatus.tsx:94]: everything seems to still work correctly from my testing

**Pull request merged**
